cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/hdfs/HdfsStorage.java

```diff
@@ -36,4 +37,22 @@ public StorageType.Type getType() {
   public StorageClient<?> getClient() {
     return hdfsStorageClient;
   }
+
+  /**
+   * Allocates Table Space for the HDFS storage.
+   *
+   * <p>tableLocation looks like: /{rootPrefix}/{databaseId}/{tableId}-{tableUUID} We strip the
+   * endpoint as it's not needed for HDFS.
+   *
+   * @return the table location
+   */
+  @Override
+  public String allocateTableSpace(
+      String databaseId,
+      String tableId,
+      String tableUUID,
+      String tableCreator,
+      boolean skipProvisioning) {
+    return Paths.get(getClient().getRootPrefix(), databaseId, tableId + "-" + tableUUID).toString();
```

Comment:
This concatenation logic could also go into the TableLocation class. Why do we not need an endpoint for HDFS and local filesystem? Is this because we are relying on default filesystem in configuration? It seems like we want to be consistent here at the Storage interface level and state that the table location returned is either absolute or relative. I am guessing we probably want the former.

Also wondering if we would want to use URI.resolve instead of Paths class to keep consistency? Paths class assumes a default filesystem. I am not entirely sure of the complete behavior of URI.resolve with two relative URIs though.

Reply:
> Why do we not need an endpoint for HDFS and local filesystem? Is this because we are relying on default filesystem in configuration? It seems like we want to be consistent here at the Storage interface level and state that the table location returned is either absolute or relative. I am guessing we probably want the former.

We are doing this because of backward compatibliity reasons. Ideally we go to a state towards absolute but because this code gets used in li-openhouse we've to be careful. 
I plan to do as follows:
- openhouse: /db/tb, li-openhouse: <missing Li-HDFSStorage>
- openhouse: /db/tb, li-openhouse: /db/tb
- openhouse: hdfs://db/tb,  li-openhouse: db/tbl

>Also wondering if we would want to use URI.resolve instead of Paths class to keep consistency? Paths class assumes a default filesystem. I am not entirely sure of the complete behavior of URI.resolve with two relative URIs though.

Ack, let me give it a shot. I kept the implementation backward compatible. I do see a value in just aligning towards URI.

Reply:
Discussed offline, summary: https://github.com/linkedin/openhouse/pull/119#discussion_r1631640467